src/coreclr/nativeaot/System.Private.CoreLib/src/System/AppContext.NativeAot.cs

```diff
@@ -1,13 +1,34 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Generic;
 using System.Runtime;
 using System.Runtime.ExceptionServices;
+using System.Text;
 
 namespace System
 {
     public static partial class AppContext
     {
+        private static unsafe Dictionary<string, object?> InitializeDataStore()
+        {
+            uint count = RuntimeImports.RhGetKnobValues(null, null);
+
+            byte*[] keys = new byte*[count];
+            byte*[] values = new byte*[count];
+            RuntimeImports.RhGetKnobValues(keys, values);
```

Comment:
```suggestion
            uint count = RuntimeImports.RhGetKnobValues(out byte* keys, out byte* values);
```
Avoid the intermediate array?

Reply:
Nice simplification!